modules/plugin/feature-pregeneralized/src/test/java/org/geotools/data/gen/PreGeneralizedFeatureSourceNotCachedTest.java

```diff
@@ -0,0 +1,75 @@
+package org.geotools.data.gen;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.spy;
+
+import org.geotools.data.Query;
+import org.geotools.data.Repository;
+import org.geotools.data.Transaction;
+import org.geotools.util.factory.Hints;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class PreGeneralizedFeatureSourceNotCachedTest
+        extends AbstractPreGeneralizedFeatureSourceTest {
+
+    static final String ConfigName = "src/test/resources/geninfo_shapefile.xml";
+    private Repository repository;
+
+    @Override
+    protected Repository getRepository() {
+        repository = spy(new DSFinderRepository());
+        return repository;
+    }
+
+    @Test
+    public void getBoundsHitsRepositoryDataStore() {
+        try {
+            PreGeneralizedDataStore ds = getDataStore(ConfigName);
+            PreGeneralizedFeatureSource fs =
+                    (PreGeneralizedFeatureSource) ds.getFeatureSource("GenStreams");
+
+            fs.getBounds();
```

Comment:
Can you run the getBounds once more, after this call, and verify this will also make the store fetch from the repository?

Reply:
We can do that ... Will call it again